tests/run-make/incr-foreign-head-span/Makefile

```diff
@@ -15,7 +15,7 @@ INCR=$(TMPDIR)/incr
 all:
 	cp first_crate.rs second_crate.rs $(TMPDIR)
 	$(RUSTC) $(TMPDIR)/first_crate.rs  -C incremental=$(INCR) --target $(TARGET) --crate-type lib
-	$(RUSTC) $(TMPDIR)/second_crate.rs -C incremental=$(INCR) --target $(TARGET)  --extern first-crate=$(TMPDIR) --crate-type lib
+	$(RUSTC) $(TMPDIR)/second_crate.rs -C incremental=$(INCR) --target $(TARGET)  --extern first_crate=$(TMPDIR)/libfirst_crate.rlib --crate-type lib
```

Comment:
I'm still baffled that this test used to work at all, can somebody enlighten me? If I manually reproduce the steps taken by this Makefile, I correctly get the error *can't find crate for `first_crate`* with a nightly rustc (since the name is *first-crate* on master). Is there any preprocessing going on here?

Apart from the incorrect crate name, `--extern` can't point to a directory, which I've also fixed. What is going on?

Reply:
This is weird indeed. I've checked the logs of recently successful runs and `tests/run-make/incr-foreign-head-span` *does* succeed on master. Very weird.